blueocean-dashboard/src/main/java/io/jenkins/blueocean/preload/FavoritesStatePreloader.java

```diff
@@ -0,0 +1,83 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package io.jenkins.blueocean.preload;
+
+import hudson.Extension;
+import hudson.model.User;
+import io.jenkins.blueocean.commons.BlueUrlTokenizer;
+import io.jenkins.blueocean.commons.RESTFetchPreloader;
+import io.jenkins.blueocean.commons.stapler.ModelObjectSerializer;
+import io.jenkins.blueocean.rest.Reachable;
+import io.jenkins.blueocean.rest.model.BlueFavorite;
+import io.jenkins.blueocean.rest.model.BlueFavoriteContainer;
+import io.jenkins.blueocean.service.embedded.rest.UserImpl;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Preload user favorites.
+ *
+ * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
+ */
+@Extension
+public class FavoritesStatePreloader extends RESTFetchPreloader {
+
+    private static final Logger LOGGER = Logger.getLogger(FavoritesStatePreloader.class.getName());
+
+    @Override
+    protected FetchData getFetchData(@Nonnull BlueUrlTokenizer blueUrl) {
+        User jenkinsUser = User.current();
+
+        if (jenkinsUser != null) {
+            UserImpl blueUser = new UserImpl(jenkinsUser);
+            BlueFavoriteContainer favoritesContainer = blueUser.getFavorites();
+
+            if (favoritesContainer != null) {
+                JSONArray favorites = new JSONArray();
+                Iterator<BlueFavorite> favoritesIterator = favoritesContainer.iterator();
+
+                while(favoritesIterator.hasNext()) {
+                    Reachable favorite = favoritesIterator.next();
+                    try {
+                        favorites.add(JSONObject.fromObject(ModelObjectSerializer.toJson(favorite)));
+                    } catch (IOException e) {
+                        LOGGER.log(Level.FINE, String.format("Unable to preload favorites for User '%s'. Serialization error.", jenkinsUser.getFullName()), e);
+                        return null;
+                    }
+                }
+
+                return new FetchData(favoritesContainer.getLink().getHref(), favorites.toString());
+            }
+        }
+
+        // Don't preload any data on the page.
+        return null;
+    }
```

Comment:
One thing I noticed in the different preloaders is that they have a similar code structure in `getFetchData` where they check if they should be active, build up data if they should, otherwise return null. From API standpoint, I wonder if `RESTFetchPreloader` should have an additional method like:

`boolean shouldPreload(BlueUrlTokenizer blueUrl)`

Which determines whether the extension actually does anything. It would give a little structure to that logic check, and maybe help remind people that they should be making an explicit check in that method to avoid someone accidentally writing a preloader that fires for all URL's.

If you decided that was worthwhile, then you'd just need a tweak in `RESTFetchPreloader.getStateJson()` which calls that method first and only proceeds if it returns true. 

I don't think it's a big deal, but it might result in a small improvement in the API (and slightly less nesting of if statements in `getFetchData`)